### PR TITLE
Update core doc

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Update core component developer documentation

### Tickets:
 - NA

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: src/core/docs/api_details.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# OpenVINO Core API
+
+OpenVINO Core API contains two folders:
+ * [ngraph](../include/ngraph/) - is a legacy API, this API is not developed more. Only aliases to new operation and operation sets extend this API.
```
Comment (tsavina) [inline src/core/docs/api_details.md:4]: is not developed more ---> is no longer being developed
maybe "aliases to new operationS"?

File: src/core/docs/debug_capabilities.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# OpenVINO Core debug capabilities
+
+OpenVINO Core contains a set of different debug capabilities which allow to make developer live a little bit easier.
```
Comment (tsavina) [inline src/core/docs/debug_capabilities.md:3]: allow to make ---> allow making
 live  ---> life 
 overall pretty useless sentence, can we elaborate on how exactly it makes it easier or just remove it?

File: src/core/docs/api_details.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * [openvino](../include/openvino/) - current public API, this part will be described below.
```
Comment (tsavina) [inline src/core/docs/api_details.md:5]:
```suggestion
 * [openvino](../include/openvino/) - current public API, this part is described below.
```

File: src/core/docs/api_details.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    <a href="../include/openvino/op/">op/</a>                     // Contains all supported OpenVINO operations
+    <a href="../include/openvino/opsets/">opsets/</a>                 // Contains definitions of each official OpenVINO opset
+    <a href="../include/openvino/pass/">pass/</a>                   // Defines classes for developing transformation and several common transformations
+    <a href="../include/openvino/runtime/">runtime/</a>                // Contains OpenVINO tensor definition
+ </code>
+</pre>
+
+## Main structures for model representation
+
+* `ov::Model` is located in [openvino/core/model.hpp](../include/openvino/core/model.hpp) and provide API for model representation. For more details please read [OpenVINO Model Representation Guide](https://docs.openvino.ai/latest/openvino_docs_OV_UG_Model_Representation.html).
```
Comment (tsavina) [inline src/core/docs/api_details.md:21]:
```suggestion
* `ov::Model` is located in [openvino/core/model.hpp](../include/openvino/core/model.hpp) and provides API for model representation. For more details, read [OpenVINO Model Representation Guide](https://docs.openvino.ai/latest/openvino_docs_OV_UG_Model_Representation.html).
```

File: src/core/docs/api_details.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+* `ov::Node` is a base class for all OpenVINO operations, the class is located in the [openvino/core/node.hpp](../include/openvino/core/node.hpp).
+* `ov::Shape` and `ov::PartialShape` classes represents shapes in the OpenVINO, these classes are located in the [openvino/core/shape.hpp](../include/openvino/core/shape.hpp) and [openvino/core/partial_shape.hpp](../include/openvino/core/partial_shape.hpp) respectively. For more information please read [OpenVINO Shapes representation](./shape_propagation.md#openvino-shapes-representation).
```
Comment (tsavina) [inline src/core/docs/api_details.md:23]:
```suggestion
* `ov::Shape` and `ov::PartialShape` classes represent shapes in the OpenVINO, these classes are located in the [openvino/core/shape.hpp](../include/openvino/core/shape.hpp) and [openvino/core/partial_shape.hpp](../include/openvino/core/partial_shape.hpp) respectively. For more information, read [OpenVINO Shapes representation](./shape_propagation.md#openvino-shapes-representation).
```

File: src/core/docs/api_details.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+* `ov::element::Type` class represents element type for OpenVINO Tensors and Operations. The class is located in the [openvino/core/type/element_type.hpp](../include/openvino/core/type/element_type.hpp).
+* `ov::Tensor` is used for memory representation inside the OpenVINO. The class is located in the [openvino/runtime/tensor.hpp](../include/openvino/runtime/tensor.hpp).
```
Comment (tsavina) [inline src/core/docs/api_details.md:25]:
```suggestion
* `ov::Tensor` is used for memory representation inside OpenVINO. The class is located in the [openvino/runtime/tensor.hpp](../include/openvino/runtime/tensor.hpp).
```

File: src/core/docs/debug_capabilities.md (hunk 2 above)
Comment (ilyachur) [reply src/core/docs/debug_capabilities.md]: Changed sentence.

**Comment by ilyachur** [comment]
@tsavina all comments were fixed

File: src/core/docs/debug_capabilities.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# OpenVINO Core debug capabilities
+
+OpenVINO Core contains a set of different debug capabilities which allow making developer life a little bit easier, because these capabilities support to collect information about object statuses in the process of OpenVINO Runtime execution and report this information to developer.
```
Comment (tsavina) [inline src/core/docs/debug_capabilities.md:3]:
```suggestion
OpenVINO Core contains a set of different debug capabilities that make developer life easier by collecting information about object statuses during OpenVINO Runtime execution and reporting this information to the developer.
```

**Review by tsavina — APPROVED** [review]

**Pull request merged**